### PR TITLE
fix(markers): improve reset groups operations

### PR DESCRIPTION
**Pull request by wpanas**
- fix function _resetGroup
- add function _resetUnusedGroups
- add function _existDomContainer

### CODE REVIEW

**Comment by wpanas** [comment]
@nmccready I forgot to run "grunt" before last push. Should I make another push?

File: src/services/leafletMarkersHelpers.js [hunk 1]
```diff
@@ -101,8 +106,8 @@ angular.module('ui-leaflet').service('leafletMarkersHelpers', function ($rootSco
     };
 
     var _resetMarkerGroup = function (groupName) {
-        if (isDefined(groups[groupName])) {
-            groups.splice(groupName, 1);
+        if (angular.isDefined(groups[groupName])) {
```
Comment (nmccready) [inline src/services/leafletMarkersHelpers.js]: What is wrong with isDefined here? It ultimately points to the same thing with less typing.

File: src/directives/markers.js [hunk 2]
```diff
@@ -166,7 +166,7 @@ angular.module('ui-leaflet').directive('markers',
                 //ie the options to only check !== (reference check) instead of angular.equals (slow)
                 newMarker = markerModels[name];
                 oldMarker = oldMarkerModels[name];
-                equals = angular.equals(newMarker,oldMarker) && isEqual;
+                equals = isEqual && angular.equals(newMarker, oldMarker);
```
Comment (nmccready) [inline src/directives/markers.js]: :8ball:

File: src/services/leafletMarkersHelpers.js (hunk 1 above)
Comment (wpanas) [inline src/services/leafletMarkersHelpers.js]: Yes, it does. But it seems to be unclear whether isDefined was defined in this scope or is a global function. Using angular.isDefined is obvious that we use isDefined from global object angular.

File: src/directives/markers.js (hunk 2 above)
Comment (wpanas) [inline src/directives/markers.js]: Sorry, I don't get this one.

File: src/services/leafletMarkersHelpers.js (hunk 1 above)
Comment (nmccready) [inline src/services/leafletMarkersHelpers.js]: It is in scope with the function. Just stick to the patterns that the library already uses please. Which is making vars in scope from leafletHelpers. There is very little going on in terms of global scope. Like you said global scope is normally either L or angular thats it.

Comment (wpanas) [inline src/services/leafletMarkersHelpers.js]: Done.

File: src/directives/markers.js (hunk 2 above)
Comment (nmccready) [inline src/directives/markers.js]: It means clutch / good job.

**Comment by nmccready** [comment]
LGTM

**Pull request merged**